### PR TITLE
Adding a new option to ignore the inherit prefix in @Controller annotation

### PR DESCRIPTION
**Pull request by somoza**
If you have this prefix at the top of your class:

` * @Controller(prefix="/your/prefix")`

You can avoid it using no_prefix parameter in your routes annotations:

`* @Get("/list", as="your.route.index", no_prefix="true")`

On this way, the route gonna be /list instead /your/prefix/list

### CODE REVIEW

File: src/Routing/Annotations/Annotations/Controller.php [hunk 1]
```diff
@@ -34,7 +34,9 @@ public function modifyCollection(EndpointCollection $endpoints, ReflectionClass
     protected function prefixEndpoints(EndpointCollection $endpoints)
     {
         foreach ($endpoints->getAllPaths() as $path) {
-            $path->path = $this->trimPath($this->prefix, $path->path);
+            if(!$path->no_prefix) {
```
Comment (loonytoons) [inline src/Routing/Annotations/Annotations/Controller.php]: Can you move the `if` to before the `foreach`? No point in looping if we don't need to.
Also please add a space between the `if` and the `(`

**Comment by loonytoons** [comment]
Awesome, thanks @somoza, looks like a useful change.
Can you add some tests to cover this functionality? I'm slowly trying to build up the test coverage so keen to have all new changes covered. Thanks :-)

File: src/Routing/Annotations/Annotations/Controller.php (hunk 1 above)
Comment (somoza) [reply src/Routing/Annotations/Annotations/Controller.php]: Thanks for your answer. The no_prefix is added (or not) in every route, so it's mandatory to check it in each iteration...

**Comment by somoza** [comment]
How can I add tests to cover this? Please let me know and I can build something...

File: src/Routing/Annotations/Annotations/Controller.php (hunk 1 above)
Comment (somoza) [reply src/Routing/Annotations/Annotations/Controller.php]: Space added as you request :)

Comment (loonytoons) [reply src/Routing/Annotations/Annotations/Controller.php]: Oh doh, totally misread that check 😂

**Comment by loonytoons** [comment]
> How can I add tests to cover this? Please let me know and I can build something...

So with the tests, we have some test controllers in `annotations/tests/Routing/fixtures/annotations/`, and some expected scanned results in `annotations/tests/Routing/results/`, then in the `RoutingAnnotationScannerTest.php` there's a couple of checks to ensure that the scan and details scan outputs the expected values.

So I'd create a new controller and expected results set, and ensure that for a controller with a prefix, that one method includes the prefix but the one with `no_prefix` doesn't

**Review by somoza — COMMENTED** [review]
Unit testing added

**Comment by loonytoons** [comment]
awesome, thanks @somoza!

**Pull request merged**